nx-dev/data-access-documents/src/lib/blog.api.ts

```diff
@@ -143,4 +150,50 @@ export class BlogApi {
       }
     }
   }
+
+  private fileExists(filePath: string): boolean {
+    try {
+      accessSync(filePath, constants.F_OK);
+      return true;
+    } catch (error) {
+      return false;
+    }
+  }
+
+  private determineOgImage(imagePath: string): {
+    image: string;
+    type: string;
+  } {
+    const allowedExtensions = ['.png', '.webp', '.jpg', '.jpeg'];
+    const defaultImage = 'https://nx.dev/socials/nx-media.png';
+    const defaultType = 'png';
+
+    if (!imagePath) {
+      return { image: defaultImage, type: defaultType };
+    }
+    const { ext } = parse(imagePath);
+
+    if (!allowedExtensions.includes(ext)) {
+      const foundExt = allowedExtensions.find((allowedExt) => {
```

Comment:
I'd love if this at some point would automatically run a conversion. But this def works for now